libs/cdk/src/lib/hover/hover.directive.ts

```diff
@@ -0,0 +1,75 @@
+import { ConnectionPositionPair, Overlay } from '@angular/cdk/overlay';
+import { TemplatePortal } from '@angular/cdk/portal';
+import { Directive, ElementRef, HostListener, inject, Input, TemplateRef, ViewContainerRef } from '@angular/core';
+
+/**  Interface for the hover context */
+export interface HoverContext<T = unknown> {
+  /** defining an implicit type */
+  $implicit: T;
+}
+
+/** Defining a directive to be used across the application */
+@Directive({
+  /**  Selector for using the directive */
+  selector: '[hraHover]',
+  /** Defining the directive to be standalone */
+  standalone: true,
+})
+export class HoverDirective<T = unknown> {
+  /** Input for the template portal */
+  @Input('hraHover')
+  set content(content: TemplateRef<HoverContext<T>>) {
+    this.portal = new TemplatePortal(content, this.viewContainerRef);
+    this.updateContent();
+  }
+
```

Comment:
Add an input called `hraHoverData` and update the template's data when it changes